lib/rubocop/cop/lint/useless_return.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Lint
+      # This cop checks for the use of a return in a context where it will
+      # be ignored. (both in initialize and setter methods)
+      #
+      # @example
+      #
+      #   # bad
+      #   def initialize
+      #     foo
+      #     return :qux if bar?
+      #     baz
+      #   end
+      #
+      #   def foo=(bar)
+      #     return 42
+      #   end
+      #
+      class UselessReturn < Cop
+        include OnMethodDef
+
+        MSG = 'Do not use return in `%s`.'.freeze
+        def on_method_def(node, method_name, _args, _body)
```

Comment:
I would probably use `#on_return` as the entry point, and then look for an enclosing method definition. WDYT?